stackit/internal/utils/utils.go

```diff
@@ -0,0 +1,62 @@
+package utils
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-framework/types/basetypes"
+
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+// ReconcileStringLists reconciles two string lists by removing elements from the
+// first list that are not in the second list and appending elements from the
+// second list that are not in the first list.
+// This preserves the order of the elements in the first list that are also in
+// the second list, which is useful when using ListAttributes in Terraform.
+// The source of truth is the second list.
+func ReconcileStringLists(list1, list2 []string) []string {
```

Comment:
```suggestion
func ReconcileStringSlices(slice1, slice2 []string) []string {
```
More accurate like this